yaptide/celery/utils/pymc.py

```diff
@@ -112,23 +115,46 @@ def __call__(self, file_path: str, rng_seed: int) -> None:
     update_fluka_function(str(input_file.resolve()), random_seed)
 
 
-def execute_simulation_subprocess(dir_path: Path, command_as_list: list[str]) -> tuple[bool, str, str]:
+def execute_simulation_subprocess(dir_path: Path, command_as_list: list[str], celery_id: str,
+                                  sim_type: str) -> tuple[bool, str, str]:
     """Function to execute simulation subprocess."""
     process_exit_success: bool = True
     command_stdout: str = ""
     command_stderr: str = ""
+    state: str = ""
     try:
-        completed_process = subprocess.run(command_as_list,
-                                           check=True,
-                                           cwd=str(dir_path),
-                                           stdout=subprocess.PIPE,
-                                           stderr=subprocess.PIPE,
-                                           text=True)
-        logging.info("simulation subprocess with return code %d finished", completed_process.returncode)
-
-        # Capture stdout and stderr
-        command_stdout = completed_process.stdout
-        command_stderr = completed_process.stderr
+        process = subprocess.Popen(command_as_list,
+                                   cwd=str(dir_path),
+                                   stdout=subprocess.PIPE,
+                                   stderr=subprocess.PIPE,
+                                   text=True)
+        try:
+            # wait untill simulation subprocess is not marked in celery as ended
+            while not state.get("end", False):
+                time.sleep(1)
+                # Check if the task has been marked to dump
+                simulation = AsyncResult(celery_id)
```

Comment:
do you need to get simulation object inside the loop ? maybe it would be enough to get it once outside the loop and just call `get_task_meta` inside.